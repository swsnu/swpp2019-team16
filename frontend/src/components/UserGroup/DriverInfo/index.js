import DriverInfo from './DriverInfo';
export default DriverInfo;
