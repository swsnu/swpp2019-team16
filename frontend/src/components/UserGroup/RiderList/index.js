import RiderList from './RiderList';
export default RiderList;
