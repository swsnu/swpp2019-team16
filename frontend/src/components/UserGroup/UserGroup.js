import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Heading from '../common/Heading/Heading';
import Container from '../common/Container/Container';
import Button from '../common/Button/Button';
import Typography from '@material-ui/core/Typography';

const UserGroupBlock = styled.div``;

UserGroup.propTypes = {
  onClick: PropTypes.func,
  googleMap: PropTypes.object,
  driverInfo: PropTypes.string.isRequired,
};

function UserGroup({ onClick, googleMap, driverInfo }) {
  return (
    <UserGroupBlock>
      <Container fixed children={<Heading title="My Group" />} />
      <Container>
        <Typography variant="h5">This is where Google Map will be</Typography>
        {googleMap}
      </Container>
      <Container>
        <Typography variant="h5">Driver Info:</Typography>
        {driverInfo}
      </Container>
      <Container>
        <Typography variant="h5">Time left until departure:</Typography>
      </Container>
      <Container maxwidth="sm">
        <Button
          variant="contained"
          onClick={onClick}
          children="On Taxi"
          fullwidth="false"
        />
      </Container>
    </UserGroupBlock>
  );
}

export default UserGroup;
