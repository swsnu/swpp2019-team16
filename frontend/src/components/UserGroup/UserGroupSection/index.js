import UserGroupSection from './UserGroupSection';
export default UserGroupSection;
