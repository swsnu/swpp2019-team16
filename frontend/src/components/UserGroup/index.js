import UserGroup from './UserGroup';
export default UserGroup;
