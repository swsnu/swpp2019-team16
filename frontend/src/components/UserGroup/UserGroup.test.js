import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import UserGroup from './UserGroup';

describe('UserGroup />', () => {
  it('SHOULD match with snapshot', async () => {
    const driverInfo = 'mock info';

    const { container } = render(
      <UserGroup onclick={() => {}} driverInfo={driverInfo} />,
    );
    expect(container).toMatchSnapshot();
  });
});
