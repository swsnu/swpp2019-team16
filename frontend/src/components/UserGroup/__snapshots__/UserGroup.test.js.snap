// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGroup /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH uTxCW"
  >
    <main
      class="sc-bdVaJa ekJyAd"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
        title="My Group"
      >
        My Group
      </h2>
    </main>
    <main
      class="sc-bdVaJa ekJyAd"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        This is where Google Map will be
      </h5>
    </main>
    <main
      class="sc-bdVaJa ekJyAd"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Driver Info:
      </h5>
      mock info
    </main>
    <main
      class="sc-bdVaJa ekJyAd"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Time left until departure:
      </h5>
    </main>
    <main
      class="sc-bdVaJa ekJyAd"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-contained"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          On Taxi
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </main>
  </div>
</div>
`;
